core/src/query.rs

```diff
@@ -59,90 +60,113 @@ impl MongoQuery {
         client: &MongoConnection,
         db: &Database,
     ) -> Result<TranslateCommandResponse> {
-        let schema_collection = db.collection::<Document>("__sql_schemas");
-
-        let collection_names = namespaces
-            .iter()
-            .map(|namespace| namespace.collection.as_str())
-            .collect::<Vec<&str>>();
-
-        // Create an aggregation pipeline to fetch the schema information for the specified collections.
-        // The pipeline uses $in to query all the specified collections and projects them into the desired format:
-        // "dbName": { "collection1" : "Schema1", "collection2" : "Schema2", ... }
-        let schema_catalog_aggregation_pipeline = vec![
-            doc! {"$match": {
-                "_id": {
-                    "$in": collection_names
+        let collection_names: Vec<String> = client.runtime.block_on(async {
+            db.list_collection_names()
+                .await
+                .map_err(Error::QueryExecutionFailed)
+        })?;
+
+        let sql_schemas_collection_exists =
+            collection_names.contains(&SQL_SCHEMAS_COLLECTION.to_string());
+
+        if !sql_schemas_collection_exists {
```

Comment:
That's not quite true. If we pass in an empty schema, it will work "sometimes". For example, I think that `select a,b,c from mytable` works even without schema.

Reply:
Yeah that's why I wrote "so **most** queries will fail.", which I think is correct? What would you like me to change it to?

Reply:
ah, I see what you meant now. Maybe we can be a bit more direct and not confuse users with the schema collection name.
What about "There are no schema information in database {0}. SQL capabilities will be very limited. Please make sure to add schema information before querying" or maybe "There are no schema information in database {0}. SQL capabilities will be very limited. Please make sure to add schema information before using the driver"?